### ISSUE TITLE
GM API version 1.0.1: removeElement() shows unexpected behavior when called from el_changed callback 

### ISSUE THREAD

**Issue by LaxmiBhavaniM** [eweitnauer/gm-play-iub-2017#74]
Following is a sample html document in which i tried to check the functionality of cmodel.removeElement() in GM version 1.0.1.

```<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8">
        <title>title</title>
        <script src="https://graspablemath.com/shared/libs/gmath/gm-inject.js"></script>
    </head>
    <body>
        <div id='gm-div' style="margin: 20px; height: 400px"></div>
        <button onclick="clearCanvas()">Clear Canvas</button>
    </body>
        <script>
            var canvas=null;
            loadGM(initCanvas, { version: '1.0.1' });

            function initCanvas() {
              canvas = new gmath.Canvas('#gm-div' , {use_toolbar: false, vertical_scroll: false });
              canvas.model.createElement('derivation', { eq: '2x+1=3', pos: { x: 'center', y: 50 } });
                    canvas.model.on('el_changed', function(evt) {
                        if(evt.last_eq.startsWith("x=")){
                            console.log("matched");
                            clearCanvas();
                        }

                    });

            }

            function clearCanvas() {
                //delete all elements on GM canvas
                while(canvas.model.elements().length>0){
                    canvas.model.removeElement(canvas.model.elements()[0]);
                }
               // Add new equation
                canvas.model.createElement('derivation', { eq: '5x+5=5', pos: { x: 'center', y: 50 } });
            }
                            
                            
        
        
        </script>
</html>
```
When the page loads,
![image](https://cloud.githubusercontent.com/assets/15314579/23849493/2d66bcf8-07b2-11e7-9651-34033f1b311e.png)

The clearCanvas() function works fine when it called by clicking on the button.
But, when  clearCanvas() is called from the el_change callback function, the canvas equation is cleared for a brief amount of time and reappears on the canvas when mouse is released. 

This is how it looks when the function is called from el_changed callback:
![image](https://cloud.githubusercontent.com/assets/15314579/23849860/82e94924-07b3-11e7-8276-433dc6b477b9.png)

As a result, there two overlapping elements on canvas and this is causing an issue when we upgraded the GM version to 1.0.1 in the game. Are we doing it right?

**Comment by eweitnauer** [comment]
I'll look into this, thank you for reporting the bug!

**Comment by eweitnauer** [comment]
I will upload a fixed version and some more detailed API docs on this later today! Please check and update the code accordingly when I made the changes.

**Comment by eweitnauer** [comment]
The bug is fixed in version 1.0.4!

The reason was that the event was triggered before the interaction was finished, so you were removing the element while the user was still dragging terms.

There now is an "el_changed" and an "el_change" event. For the egg-solving, you want to use the "el_changed" event. If you want to clear the canvas, please use `canvas.controller.reset()`. For matching the rocks I think it would be nice to actually use the "el_change" event, so the rock gets destroyed as soon as you are in the right state, even before you drop the term.

Please take a look at the updated API docs and let me know if this works for you!

**Comment by LaxmiBhavaniM** [comment]
Thank you @eweitnauer . The game works well with the new GM version.
